arelle/XmlValidate.py

```diff
@@ -510,19 +541,19 @@ def validateValue(modelXbrl, elt, attrTag, baseXsdType, value, isNillable=False,
                             month, day, zSign, zHrMin, zHr, zMin = match.groups()
                             if int(day) > {2:29, 4:30, 6:30, 9:30, 11:30, 1:31, 3:31, 5:31, 7:31, 8:31, 10:31, 12:31}[int(month)]:
                                 raise ValueError("invalid day {0} for month {1}".format(day, month))
-                            xValue = gMonthDay(month, day)
+                            xValue = gMonthDay(cast(int, month), cast(int, day))
```

Comment:
Preexisting: the __init__ param types for `gYearMonth`, `gMonthDay`, `gYear`, `gMonth`, and `gDay` are wrong. They should all accept strings which would allow us to avoid the casts here.